### PR TITLE
[botan] Update, higher level maintainer functions

### LINKED ISSUE #27319: [botan] Build error
**Host Environment**

- OS: Windows 10
- Compiler: MSVC vs2017

**To Reproduce**

vcpkg install botan

**Failure logs**

- Cut and paste the appropriate build messages from the console output.

CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: C:/src/vcpkg/downloads/tools/jom/jom-1.1.3/jom.exe /J9
    Working Directory: C:/src/vcpkg/buildtrees/botan/x86-windows-rel
    See logs for more information:
      C:\src\vcpkg\buildtrees\botan\build-x86-windows-rel-out.log
      C:\src\vcpkg\buildtrees\botan\build-x86-windows-rel-err.log

Call Stack (most recent call first):
  ports/botan/portfile.cmake:112 (vcpkg_execute_build_process)
  ports/botan/portfile.cmake:134 (BOTAN_BUILD)
  scripts/ports.cmake:147 (include)

error: building botan:x86-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+botan
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[botan]+Build+error
Include '[botan] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
    vcpkg-scripts version: ce99c947b 2022-10-18 (2 hours ago)

- Attach ***every failure log file*** mentioned in the console output.

build-x86-windows-rel-err.log and build-x86-windows-rel-out.log are attached:
[build-x86-windows-rel-err.log](https://github.com/microsoft/vcpkg/files/9818984/build-x86-windows-rel-err.log)
[build-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/9818985/build-x86-windows-rel-out.log)

**Additional context**

thanks. 

Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Update to 2.19.2.
  Improve port maintainability.
  Fixes #27319.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline].
  uwp disabled, cf. https://github.com/microsoft/vcpkg/pull/27750#issuecomment-1312352368
  Given the proprietary build system, only vcpkg's official triplets are fully supported, plus mingw.
  ios, android, emscripten, bsd will probably need more changes, but it should be straightforward to add if someone can verify the result.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
uwp is now marked unsupported, because it would need patches to at least two components, and I don't want to deal with this in this PR.
- The win32 entropy component uses desktop api. It probably must be disabled, which would be a simple build system patch.
- The system RNG component has another build problem, fixed upstream for 3.0.0

So uwp should be revisited for botan 3.0.0.

**Review by LilyWangLL — DISMISSED** [review]

**Review by BillyONeal — CHANGES_REQUESTED** [review]
This breaks `botan[zlib]` on Windows: https://github.com/randombit/botan/issues/3066

Given that it was always broken, I don't think you need to fix the broken, but it would be nice. I think it should at least replicate the old broken though.

File: ports/botan/configure [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply ports/botan/configure:1]: (Sorry, to clarify, I wrote this comment when I thought I was going to merge it immediately. I am happy with the change overall except for the zlib feature being broken on Windows. Ideally upstream gives us what they want us to do?)

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/botan/configure:1]: This script is just glue between `vcpkg_configure_make`  and `configure.py`. It is a copy from what I have in the openssl non-windows PR. It becomes obsolete when there is another way to run a custom configure command in our `make` command family.

**Comment by dg0yt** [comment]
The new patch doesn work for in-source builds, but this is not relevant for vcpkg.

**Comment by BillyONeal** [comment]
> in-source builds

Why do people do that to themselves?! 😅

**Comment by dg0yt** [comment]
And why yet another proprietary configuration system...

**Review by BillyONeal — APPROVED** [review]
Thanks for also fixing the zlib issue!

**Pull request merged**